python/pyspark/sql/connect/client.py

```diff
@@ -299,7 +301,12 @@ def userAgent(self) -> str:
             raise SparkConnectException(
                 f"'user_agent' parameter should not exceed 2048 characters, found {len} characters."
             )
-        return user_agent
+        return " ".join([
```

Comment:
Don't have a strong opinion but wonder if we should return a dictionary or at least JSON string in case it's programmatically used

Reply:
This is the IETF standard format for user agent.